### PR TITLE
Update `@babel/parser` to 7.15.2

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Release note https://github.com/babel/babel/releases/tag/v7.15.2

## Checklist

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/parse/babel.js [hunk 1]
```diff
@@ -77,7 +77,6 @@ function parseWithOptions(parseMethod, text, options) {
   /** @type {import("@babel/parser").parse | import("@babel/parser").parseExpression} */
   const parse = require("@babel/parser")[parseMethod];
   const ast = parse(text, options);
-  // @ts-expect-error
```
Comment (sosukesuzuki) [inline src/language-js/parse/babel.js:80]: https://github.com/babel/babel/pull/13653

Comment (fisker) [reply src/language-js/parse/babel.js:80]: I recall there is something else missing, but I can't remember now.

**Review by fisker — APPROVED** [review]

File: src/language-js/parse/babel.js (hunk 1 above)
Comment (fisker) [reply src/language-js/parse/babel.js:80]: Oh, right, `attachComment` seems missing in  `ParserOptions` for 
 parseExpression, https://github.com/prettier/prettier/runs/3258106700#step:6:6, maybe you can fix that too.

Comment (sosukesuzuki) [reply src/language-js/parse/babel.js:80]: Oh I didn't know that! Thanks!

**Pull request merged**